### ISSUE TITLE
pkg builds some projects by default that it didn't used to

### ISSUE THREAD

**Issue by ocket8888** [apache/trafficcontrol#6273]
## This Bug Report affects these Traffic Control components:
None/all, depending on your point of view. This bug only affects `pkg`.

## Current behavior:
The `pkg` script, when given no arguments, will build every project listed by `-l`.

## Expected behavior:
Previously, the default build list was _not_ the same as the output of `-l`, which just listed all projects that _could_ be built. In particular, `docs`, `weasel`, and `grovetccfg_build` used to be skipped when building without listing specific projects. `weasel` takes a while, so this does represent a significant increase in build time.

However, this behavior does seem more consistent. I may even like it better than the old behavior, but for convenience I'd want to at least be able to explicitly exclude projects from the build, because I almost never want to build docs, source, grove (both regular grove and grovetccfg), or weasel.

So in conclusion, I'm not sure we want to consider this a regression bug or an intended feature change, but the bottom line is that things that used to not happen are now happening, so that's why it's labeled like this. Feel free to change it.

## Steps to reproduce:
Run `./pkg` from the repo root, notice what it builds.

**Comment by ocket8888** [comment]
"low impact" is a guess, since this needs more information.
